### PR TITLE
Describe setup process in documentation

### PR DESCRIPTION
**Pull request by veeara282**
I added a new documentation page about the setup wizard. Please add and correct material to this page as you see fit!

This page incorporates material from iiab/iiab#1573.

### CODE REVIEW

File: docs/content/doc/installation/setup.en-us.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+# Setup
+
+The first time you access the Gitea web interface, the home page will appear. 
+Clicking on "Register" or "Sign In" in the upper right corner will take you to 
+the setup page.
+
+<!-- TODO Describe setup options -->
+<!-- TODO Add screenshots -->
+
+After finishing the setup process, you will be directed to a page where you can
```
Comment (kolaente) [inline docs/content/doc/installation/setup.en-us.md:28]: Isn't the admin created directly during the install process?

Comment (techknowlogick) [reply docs/content/doc/installation/setup.en-us.md:28]: Admin creation during installation is optional.

Comment (veeara282) [reply docs/content/doc/installation/setup.en-us.md:28]: Good catch. I haven't gone through the setup process in a while.

**Comment by veeara282** [comment]
I just added screenshots of the setup process. What is the proper location for these images? Would it be in [`docs/static`](https://github.com/go-gitea/gitea/tree/master/docs/static)? Tested, yes that's it.

**Comment by veeara282** [comment]
@techknowlogick @kolaente Is this too much detail? Is there enough detail in the sections on database settings and optional settings?

Also, I've only had to use one screenshot so far. If we don't use the others, I might delete them from the branch before we merge.

File: docs/content/doc/installation/setup.en-us.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+Includes database type and other database-dependent settings. You can choose 
+from SQLite3, MySQL, PostgreSQL, and Microsoft SQL Server. (The default database 
+system is SQLite3.)
+
+## General Settings
+
+| Option                 | Required? | In `app.ini`                  |
+| ---------------------- | --------- | ----------------------------- |
+| Site Title             | required  | `APP_NAME`                    |
```
Comment (adelowo) [inline docs/content/doc/installation/setup.en-us.md]: Seeing this for the first time, I will assume values under the  ___Required___ column are supposed to be true or false

File: docs/content/doc/installation/setup.en-us.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+| SSH Server Port        | `server.SSH_PORT`           | no        |
+| Gitea HTTP Listen Port | `server.HTTP_PORT`          | yes       |
+| Gitea Base URL         | `server.ROOT_URL`           | yes       |
+| Log Path               | `log.ROOT_PATH`             | yes       |
+
+## Optional Settings
+
+Consists of:
+
+- Email Settings
```
Comment (0x5c) [inline docs/content/doc/installation/setup.en-us.md]: Much detail is given in the database section. However, where is the explanation here? Email setup is less ubiquitous than database setup, and in the context of Gitea, far more arcane.
Same for `Server and Third-Party Service Settings`

**Review by lafriks — APPROVED** [review]

File: docs/content/doc/installation/setup.en-us.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+**If the database file does not already exist, Gitea should be able to create it.**
+
+For MySQL, PostgreSQL, and Microsoft SQL Server, you need to specify the 
+server address, the database name, and the user credentials that Gitea will use 
+to access the database. **Make sure the user exists on the database server first.**
+
+## General Settings
+
+In this section, you can set the general attributes of the Gitea instance. For 
+more details, see the [configuration cheat sheet](/en-us/config-cheat-sheet/).
```
Comment (jolheiser) [inline docs/content/doc/installation/setup.en-us.md]: Can you use the Hugo `relref` function here?
https://gohugo.io/content-management/cross-references/

`[configuration cheat sheet]({{< relref "doc/advanced/config-cheat-sheet.en-us.md" >}})`

**Comment by zeripath** [comment]
Is this still a WIP?

**Comment by 0x5c** [comment]
Looking at email and third-party sections, I would assume so.

**Comment by veeara282** [comment]
@zeripath @0x5c Email is done. Third-party still needs to be worked on.

**Comment by 0x5c** [comment]
Email is one of those things that does need to be explained, as, contrary to database servers, someone needing email notifications to work will have but that one choice. (You don't need a database server to run Gitea, and will only see the benefits if you [already] have a server for other services, which also means that you most probably know what the server address, port, and db name should be.)

Good documentation is consistent, and that also goes with the extent to which it explains concepts, functions, and options. For this reason, I raise the question:  
*What is the goal here?*  
Is it to produce a "beginner's guide", or a blanket coverage of the *Setup* page's options?

Knowing the answer will help avoid creating a mix of the two, and thus, maintain consistency.

**Comment by veeara282** [comment]
@0x5c I want the documentation to be comprehensive - that is, summarize all of the options shown on the setup page while duplicating the config cheatsheet as little as possible. I also want it to be beginner-friendly so that someone with a reasonable level of prior sysadmin knowledge can understand it.

**Comment by 0x5c** [comment]
That still does not answer the question, but ignoring that;

1. Any documentation about the setup page *should not* rely on documentation about the configuration file, and even less for a non-authoritative one (the cheat-sheet is not maintained to the extent that the `app.ini` sample is [1]). *The setup page is not the configuration file.*
2. The currently proposed document is not consistent.
3. A beginner's guide and extensive documentation are mutually-exclusive concepts: one targets the elements that are required to *begin/"get started"*, while the other targets clear, concise, and complete documentation. One has to be chosen. 

The current Gitea documentation is none of clear, concise, and complete, but let's start breaking the trend instead of making the matters worse.
<br>

[1] - [Config Cheat Sheet](https://docs.gitea.io/en-us/config-cheat-sheet/), ¶3:

> The defaults provided here are best-effort (not built automatically). They are accurately recorded in app.ini.sample (`s/master/<tag|release>`).

**Pull request closed without merging**